### PR TITLE
PR: Implement support for "Leica L-Log" log encoding curve.

### PR DESCRIPTION
**Pull request by KelSolaar**
# Summary

This PR implements support for the *Leica L-Log* log encoding curve.

References #980.

# Preflight

**Code Style and Quality**

- [x] Unit tests have been implemented and passed.
- [x] Mypy static checking has been run and passed.
- [x] Pre-commit hooks have been run and passed.
- [N/A] New transformations have been added to the *Automatic Colour Conversion Graph*.
- [x] New transformations have been exported to the relevant namespaces, e.g. `colour`, `colour.models`.

**Documentation**

- [x] New features are documented along with examples if relevant.
- [x] The documentation is [Sphinx](https://www.sphinx-doc.org/en/master/) and [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant.

### CODE REVIEW

**Comment by nick-shaw** [comment]
I may be missing something, but it seems that `colour.models.log_encoding(0.18, 'L-Log')` works, but `colour.models.log_encoding_LLog(0.18)` does not.

**Comment by nick-shaw** [comment]
LLog (encoding and decoding) appears to be omitted from the  lists in `__init__.py` under `colour/models` and `colour/models/rgb`.

**Comment by zachlewis** [comment]
This little OCIO bit works as expected:
`!<LogCameraTransform> {name: Leica L-Log to Linear,  base: 10, log_side_slope: 0.27, log_side_offset: 0.6, lin_side_slope: 1.3, lin_side_offset: 0.0115, lin_side_break: 0.006, linear_slope: 8, direction: inverse}`

File: colour/models/rgb/transfer_functions/leica_llog.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+__email__ = "colour-developers@colour-science.org"
+__status__ = "Production"
+
+__all__ = [
+    "CONSTANTS_NLOG",
+    "log_encoding_LLog",
+    "log_decoding_LLog",
+]
+
+CONSTANTS_NLOG: Structure = Structure(
```
Comment (zachlewis) [inline colour/models/rgb/transfer_functions/leica_llog.py:44]: Should be CONSTANTS_LLOG

File: colour/models/rgb/transfer_functions/leica_llog.py [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+)
+"""*Leica L-Log* colourspace constants."""
+
+
+def log_encoding_LLog(
+    LSR: FloatingOrArrayLike,
+    bit_depth: Integer = 10,
+    out_normalised_code_value: Boolean = True,
+    in_reflection: Boolean = True,
+    constants: Structure = CONSTANTS_NLOG,
```
Comment (zachlewis) [inline colour/models/rgb/transfer_functions/leica_llog.py:62]: CONSTANTS_LLOG

**Review by zachlewis — CHANGES_REQUESTED** [review]
CONSTANTS_NLOG should be CONSTANTS_LLOG

**Comment by KelSolaar** [comment]
> CONSTANTS_NLOG should be CONSTANTS_LLOG

Thanks, good catch!

### Small Horror Story

I wrote the module two times and managed to lost the code failing with git stash fu, the first time it was correct as I remember changing it specifically [because the N-LOG one was not consistent](https://github.com/colour-science/colour/commit/8298063117df03ecb5acead827c33736c34c53b9).

**Comment by KelSolaar** [comment]
> LLog (encoding and decoding) appears to be omitted from the lists in `__init__.py` under `colour/models` and `colour/models/rgb`.

Good catch, I forgot that we need to explicitly import the transfer functions since I made the typing annotation changes.

**Pull request merged**